ome/scc
# Conda Bump version

This PR introduces a new command ``scc bump-version-conda`` to update the ``version`` and ``sha256`` in the ``meta.yaml`` files in the conda-* repositories. The command will be run in a GitHub action using a scheduled job in the conda-* repositories i.e. similar to a bot. If changes are detected, the ``meta`` files are modified, committed and a PR is created. Note that the method ``bump-version-conda`` does not create the PR, this done via GitHub action in order to reduce the boiler plate required to create a PR


What does this PR do?
 * Find the meta.yaml files in the conda-* repositories
 * Determine the corresponding repository if the ``--repo`` flag is not set e.g. ``conda-omero-py`` will look for ``omero-py``
 * Find the latest tag from the target repository using ``git ls-remote``. rc, dv, m* are not considered
 * Check the version in the meta file.
 * If no version changed, the process stops
 * If version changed, determine which sha256 to use depending on the package. This could be from GH, pypi, downloads.o.org
 * Update the version/sha256 in the meta.yml files
 * commit changes. 

To test this PR:
* In a virtual environment, install this branch
* Activate the environment

Test 1:
  * Go to ``conda-omero-py``:
  *  run ``scc bump-version-conda``
  *  nothing should be committed.
  *  modify the version in meta file, save, do not commit. 
  * run the command again
  * the process will run, the file will be updated but the commit will not occur
  
Test 2:
 * Go to ``conda-bftools``:
 * run ``scc bump-version-conda --repo ome/bioformats``
 * the 2 meta files should be modified: version and sha256. Check that the version is 6.7.0, and the sh256 is the value from 
   https://downloads.openmicroscopy.org/bio-formats/6.7.0/artifacts/bftools.zip.sha1
 * Changes should be committed

Test 3:
 * conda-raw2ometiff
 * run ``scc bump-version-conda --repo glencoesoftware/raw2ometiff``
 * Check that nothing is changed i.e. rc is not applied.
 
Workflow in action https://github.com/jburel/conda-omero-py/pull/4


cc @joshmoore @sbesson
